example/plugin/toerror/Readme.md

```diff
@@ -0,0 +1,55 @@
+The deriveToError function is useful for composing bool-returning functions along with traditional error-returning functions.
+
+Given the following input:
+
+```go
+package toerror
+
+import (
+	"fmt"
+	"net/http"
+)
+
+func parseMajorHTTPVersion(versionString string) (int, error) {
+	return deriveCompose(
+		deriveToError(fmt.Errorf("HTTP version parsing failed"), http.ParseHTTPVersion),
+		func(major, minor int) (int, error) {
+			return major, nil
```

Comment:
One final nit, it would be great if this example function actually also had the possibility of return an error.
For example
`if major < x then`